src/matrixeventprocessor.ts

```diff
@@ -59,12 +62,14 @@ export class MatrixEventProcessor {
     private discord: DiscordBot;
     private matrixMsgProcessor: MatrixMessageProcessor;
     private mxCommandHandler: MatrixCommandHandler;
+    private mxUserProfileCache: TimedCache<string, {displayname: string, avatar_url: string}>;
 
     constructor(opts: MatrixEventProcessorOpts, cm?: MatrixCommandHandler) {
         this.config = opts.config;
         this.bridge = opts.bridge;
         this.discord = opts.discord;
         this.matrixMsgProcessor = new MatrixMessageProcessor(this.discord);
+        this.mxUserProfileCache = new TimedCache(PROFILE_CACHE_LIFETIME);
```

Comment:
Okay, we cache the profiles now, great! Only problem.....don't we need to flush the cache for that user on `m.room.member` changes? This doesn't seem to happen anywhere?

Reply:
and/or just update to the new state we already have then